libcxxabi/src/demangle/Utility.h

```diff
@@ -121,6 +137,8 @@ class OutputBuffer {
   OutputBuffer &prepend(std::string_view R) {
     size_t Size = R.size();
 
+    notifyInsertion(/*Position=*/0, /*Count=*/Size);
```

Comment:
I think this should be called after we perform the insert, so that the callback can inspect the added bytes.

Reply:
done